source/common/tls/utility.cc

```diff
@@ -17,6 +17,10 @@ namespace Extensions {
 namespace TransportSockets {
 namespace Tls {
 
+// OpenSSL caps OBJ_obj2txt for Object Identifier at 586 bytes per
+// https://github.com/openssl/openssl/blob/master/CHANGES.md#changes-between-310-and-311-30-may-2023
+static constexpr int MAX_OID_SIZE = 256;
```

Comment:
The comment (and link) say 586 bytes, but the value is 256. Is that on purpose?

Also, I think (but not 100% sure) that the 586 byte limit is the original ASN.1 encoding, not the string form.

Reply:
Yes, that's intended. As the existing code (before this PR) uses `256`.
ref: https://github.com/envoyproxy/envoy/pull/35953/files#diff-c68fb6f56100c870230eb94f44f1c0d3c43849a1326f2622b39ce9e97b847910L258

So by converting in to a const definition, that applies to both new code (my PR) and the existing code, I'm intending to keep the same 256 size limit

> the 586 byte limit is the original ASN.1 encoding, not the string form

Re-reading the openssl code, I think you are right. Let me push another commit to update the comment string to avoid confusions.

Reply:
Actually, thinking this through more, since we use boringssl and not openssl, let's just remove that comment, unless you're sure it applies to boringssl

Reply:
Done! Checked and it's indeed openssl specific. Removed the comment to make it simpler.